consensus/src/simplex/mod.rs

```diff
@@ -4939,4 +4944,387 @@ mod tests {
+            join_all(finalizers).await;
+
+            // Verify safety: no conflicting finalizations across honest reporters
+            let honest_start = faults as usize * 2; // Each twin produces 2 reporters
+            let mut finalized_at_view: BTreeMap<View, D> = BTreeMap::new();
+            for reporter in reporters.iter().skip(honest_start) {
+                let finalizations = reporter.finalizations.lock().unwrap();
+                for (view, finalization) in finalizations.iter() {
+                    let digest = finalization.proposal.payload;
+                    if let Some(existing) = finalized_at_view.get(view) {
+                        assert_eq!(
+                            existing, &digest,
+                            "safety violation: conflicting finalizations at view {view}"
+                        );
+                    } else {
+                        finalized_at_view.insert(*view, digest);
+                    }
+                }
+            }
+
+            // Verify no invalid signatures were observed
+            for reporter in reporters.iter().skip(honest_start) {
+                let invalid = reporter.invalid.lock().unwrap();
+                assert_eq!(*invalid, 0, "invalid signatures detected");
+            }
+
+            // Ensure faults are attributable to twins
+            let twin_identities: Vec<_> = participants.iter().take(faults as usize).collect();
+            for reporter in reporters.iter().skip(honest_start) {
+                let faults = reporter.faults.lock().unwrap();
+                for (faulter, _) in faults.iter() {
+                    assert!(
+                        twin_identities.contains(&faulter),
+                        "fault from non-twin participant"
+                    );
+                }
+            }
+
+            // Ensure blocked connections are attributable to twins
+            let blocked = oracle.blocked().await.unwrap();
+            for (_, faulter) in blocked.iter() {
+                assert!(
+                    twin_identities.contains(&faulter),
+                    "blocked connection from non-twin participant"
+                );
+            }
+        });
+    }
+
+    fn test_twins<S, F>(mut fixture: F)
+    where
+        S: Scheme<PublicKey = ed25519::PublicKey>,
+        F: FnMut(&mut deterministic::Context, u32) -> Fixture<S>,
+    {
+        for strategy in [
+            Strategy::View,
+            Strategy::Fixed(3),
+            Strategy::Isolate(4),
+            Strategy::Broadcast,
+            Strategy::Shuffle,
+        ] {
+            for link in [
+                Link {
+                    latency: Duration::from_millis(10),
+                    jitter: Duration::from_millis(1),
+                    success_rate: 1.0,
+                },
+                Link {
+                    latency: Duration::from_millis(200),
+                    jitter: Duration::from_millis(150),
+                    success_rate: 0.75,
```

Comment:
With the `resolver` disabled, twins are more or less useless if they fall behind. When this happens, all we have left participating is `2f+1`. If we use our typical message success rate of 50%, it is very very unlikely all leader broadcasts + notarize votes reach their destination (i.e. all we see are nullifications).

I found that boosting this to 75% leads to these tests always passing but admit it is a bit of a "junk science".